### ISSUE TITLE
`hadolint` via `pre-commit` on macOS fails when trying to update the repo.

### ISSUE THREAD

**Issue by spkane** [hadolint/hadolint#1166]
```
$ uv run pre-commit autoupdate
...
[https://github.com/hadolint/hadolint] updating v2.13.1-beta -> v2.14.0
An unexpected error has occurred: CalledProcessError: command: ('/opt/homebrew/bin/git', '-c', 'core.useBuiltinFSMonitor=false', '-C', '/var/folders/rd/9w38b2m139d3tfvp186pkt180000gn/T/tmp7015uz0k', 'fetch', 'origin', 'HEAD', '--quiet', '--filter=blob:none', '--tags')
return code: 1
stdout: (none)
stderr:
    error: You're on a case-insensitive filesystem, and the remote you are
    trying to fetch from has references that only differ in casing. It
    is impossible to store such references with the 'files' backend. You
    can either accept this as-is, in which case you won't be able to
    store all remote references on disk. Or you can alternatively
    migrate your repository to use the 'reftable' backend with the
    following command:

        git refs migrate --ref-format=reftable

    Please keep in mind that not all implementations of Git support this
    new format yet. So if you use tools other than Git to access this
    repository it may not be an option to migrate to reftables.
Check the log at /Users/spkane/.cache/pre-commit/pre-commit.log
```

**Comment by m-ildefons** [comment]
Hi Sean,

sorry for replying so late to this, but it took me a while to figure out what exactly goes wrong here and what I can (or rather can't) do about it.

As you can see from the error message, the problem is with the git repository. And by "the git repository" I mean your local clone, since the problem only occurs when the git repository is initialized on a case-insensitive filesystem.
Git remotes can't set configuration options for the clones - this has to be done client-side. This means that I can't produce a change to the Hadolint GitHub repo that will fix your problem here, as the problem lies in the way pre-commit does the initialization of your local clone of the Hadolint repo.

You can fix your instance by locating your clone of the Hadolint repository somewhere in `~/.cache/pre-commit` or the equivalent location on your system. There you can run the recommended command `git refs migrate --ref-format=reftable`.
Alternatively you can try and delete your local clone and then run `pre-commit install` again. This should also fix the situation as pre-commit will then make a new shallow clone of the Hadolint repo.

To fix this problem permanently in all cases, there needs to be a change in the way pre-commit initializes its repositories, such that it works on case-insensitive filesystems.
